test/ts-parser.test.js

```diff
@@ -66,7 +66,8 @@ describe('TypeScript Parser Test', function() {
     }
   );
 
-  it('should report errors if es2016 apis are used with es2015 tsconfig', function() {
+  // TypeScript 3.7.x no longer reports it as an error
+  it.skip('should report errors if es2016 apis are used with es2015 tsconfig', function() {
```

Comment:
should we simply remove this test case then?

Reply:
Isn't this a matter of typescript configuration? I think it should be possible to configure TypeScript to reject ES2016 (and newer) APIs.

See https://www.typescriptlang.org/docs/handbook/compiler-options.html, the part `--lib`.

List of library files to be included in the compilation.

Possible values are: 
► ES5 
► ES6 
► ES2015 
► ES7 
► ES2016 
► ES2017 
► ES2018 
► ESNext 
► DOM 
► DOM.Iterable 
► WebWorker 
► ScriptHost 
► ES2015.Core 
► ES2015.Collection 
► ES2015.Generator 
► ES2015.Iterable 
► ES2015.Promise 
► ES2015.Proxy 
► ES2015.Reflect 
► ES2015.Symbol 
► ES2015.Symbol.WellKnown 
► ES2016.Array.Include 
► ES2017.object 
► ES2017.Intl 
► ES2017.SharedMemory 
► ES2017.String 
► ES2017.TypedArrays 
► ES2018.Intl 
► ES2018.Promise 
► ES2018.RegExp 
► ESNext.AsyncIterable 
► ESNext.Array 
► ESNext.Intl 
► ESNext.Symbol 

Note: If --lib is not specified a default list of libraries are injected. The default libraries injected are: 
► For --target ES5: DOM,ES5,ScriptHost
► For --target ES6: DOM,ES6,DOM.Iterable,ScriptHost

I think the more important question to ask is why are we rejecting ES2016 syntax? Is it because it was difficult to support it in the past? If TypeScript no longer throws now, then maybe we should change the test to verify that ES2016 syntax is allowed after the dependencies are upgraded?

Reply:
We have https://github.com/strongloop/strong-docs/blob/master/test/tsconfig.json#L9 to configure the `lib` to `es2015`. The test was there to capture the fact that TypeScript rejects the usage of non es2015 apis.

We can change the test to accept the fact that latest TypeScript does not report an error for such case. The behavior won't impact how strong docs are generated anyway.

Reply:
> then maybe we should change the test to verify that ES2016 syntax is allowed after the dependencies are upgraded?

After following the discussion here I agree with rewrite the test to reflect the fact.